component-library/component-lib/src/lib/form-components/multi-checkbox/multi-checkbox.component.ts

```diff
@@ -34,123 +38,223 @@ export class MultiCheckboxComponent implements OnInit {
-          } else if (
-            this.config.children?.findIndex((child) => {
-              return child.id === response.id;
-            }) > -1
-          ) {
-            let positive = 0;
-            let negative = 0;
-
-            this.config.children?.forEach((res) => {
-              res.formGroup.get(res.id)?.value === true
-                ? positive++
-                : negative++;
-            });
-
-            if (positive > 0 && negative > 0) {
-              this.config.parent.formGroup
-                .get(this.config.parent.id)
-                ?.patchValue(true, { emitEvent: false });
-              this.config.parent.mixed = true;
-            } else if (positive > 0 && negative == 0) {
-              this.config.parent.formGroup
-                .get(this.config.parent.id)
-                ?.patchValue(true, { emitEvent: false });
-              this.config.parent.mixed = false;
-            } else if (positive == 0 && negative > 0) {
-              this.config.parent.formGroup
-                .get(this.config.parent.id)
-                ?.patchValue(false, { emitEvent: false });
-              this.config.parent.mixed = false;
-            }
+          let children = this.config.children?.findIndex((child) => {
+            return child.id === response;
+          });
+
+          if (children > -1) {
+            this.checkError(
+              this.config?.children[children]?.formGroup?.get(response)?.status,
+              this.config?.children[children]?.formGroup || new FormGroup({}),
+              response
+            );
           }
         }
+        if (this.config?.parent?.id === response) {
+          this.checkError(
+            this.config?.parent?.formGroup?.get(response)?.status,
+            this.config?.parent?.formGroup || new FormGroup({}),
+            response
+          );
+        }
       }
     );
 
-    this.errorSub = this.multicheckboxService.multiCheckboxErrorobs$.subscribe(
-      (response) => {
-        if (response.id === this.config.parent.id) {
-          if (
-            response.event.hasOwnProperty('remove') &&
-            response.event.remove === true
-          ) {
-            this.removeResponseFromError(response);
-          } else {
-            this.config?.errorMessages.push(response.event);
-          }
-        } else if (
-          this.config.children &&
-          this.config.children?.findIndex((child) => {
-            return child.id === response.id;
-          }) > -1
-        ) {
-          if (
-            response.event.hasOwnProperty('remove') &&
-            response.event.remove === true
-          ) {
-            let newErrorMessages = new Array<IErrorPairsMutltiCheckBox>();
-
-            this.config.errorMessages.forEach((error) => {
-              if (error.id != response.id) {
-                newErrorMessages.push(error);
-              }
-            });
+    if (this.config.parent != undefined) {
```

Comment:
!== 
There's a lot of falsy compares in this code; if they are intentional, that's fine, but if not, please use false (!== instead of !=)

Reply:
I will check with Alex, since he added those.

Reply:
They should be ok with !==, they are mostly used to prevent the ?. and "object is undefined" spaghetti that you get in TS